### PR TITLE
[FEATURE] Add Dropbox support for Business accounts

### LINKED ISSUE #10385: [FEATURE] Add Dropbox support for Business accounts
**Is your feature request related to a problem? Please describe.**
Dropbox Business accounts are not currently supported by our integration today. This leads to any Dropbox business users being blocked on using the integration today.

This request is to refactor the integration to support both Business and User endpoints within the same app.
https://www.dropboxforum.com/t5/Dropbox-API-Support-Feedback/Working-with-Individual-API-and-Business-API-endpoints/td-p/394857

### Steps

1. [In Progress] - Get access to Dropbox Business Sandbox development account
2. Refactor Dropbox app to check to see if a user is a business account, and return the proper headers (e.g. 'Dropbox-API-Select-User') in all calls to user endpoints

### PR DESCRIPTION
**Pull request by jcortes**
## WHY

Resolves #10385

## Summary by CodeRabbit

- **New Features**
  - Enhanced path property definitions to include filtering based on file or folder types for various Dropbox actions.
  - Increased default maximum results for file and folder searches from 100 to 1000.

- **Bug Fixes**
  - Updated logic for creating, updating, and deleting files and folders to ensure proper type validation and filtering.

- **Performance Improvements**
  - Refined path filtering logic for improved performance in folder searches and monitoring.

- **Version Updates**
  - Incremented version numbers across multiple Dropbox components to reflect recent changes and improvements.

### CODE REVIEW

**Comment by malexanderlim** [comment]
@jcortes Can the file paths be displayed in the same way as before? e.g.
Screenshot 2024-06-03 at 3 27 16 PM

instead of 
Screenshot 2024-06-03 at 3 32 57 PM

File: components/dropbox/actions/create-update-share-link/create-update-share-link.mjs [hunk 1]
```diff
@@ -5,16 +5,20 @@ export default {
...
   props: {
     dropbox,
     path: {
       propDefinition: [
         dropbox,
-        "pathFileFolder",
+        "path",
         () => ({
-          omitRootFolder: true,
+          firtOptions: [],
```
Comment (michelle0927) [inline components/dropbox/actions/create-update-share-link/create-update-share-link.mjs:17]: I thnk this is supposed to be `initialOptions`?
```suggestion
          initialOptions: [],
```

File: components/dropbox/actions/delete-file-folder/delete-file-folder.mjs [hunk 2]
```diff
@@ -4,16 +4,20 @@ export default {
...
   props: {
     dropbox,
     path: {
       propDefinition: [
         dropbox,
-        "pathFileFolder",
+        "path",
         () => ({
-          omitRootFolder: true,
+          firtOptions: [],
```
Comment (michelle0927) [inline components/dropbox/actions/delete-file-folder/delete-file-folder.mjs:16]: I think this is supposed to be `initialOptions`?
```suggestion
          initialOptions: [],
```

File: components/dropbox/actions/move-file-folder/move-file-folder.mjs [hunk 3]
```diff
@@ -4,16 +4,20 @@ export default {
   name: "Move a File/Folder",
   description: "Moves a file or folder to a different location in the user's Dropbox [See the docs here](https://dropbox.github.io/dropbox-sdk-js/Dropbox.html#filesMoveV2__anchor)",
   key: "dropbox-move-file-folder",
-  version: "0.0.9",
+  version: "0.10",
```
Comment (michelle0927) [inline components/dropbox/actions/move-file-folder/move-file-folder.mjs:7]:
```suggestion
  version: "0.0.10",
```

**Review by michelle0927 — COMMENTED** [review]
Not sure if this needs a review since it's blocked, but I left a few comments.

**Review by michelle0927 — DISMISSED** [review]
LGTM!

**Review by malexanderlim — APPROVED** [review]
LGTM!

**Pull request merged**